### DISCUSSION TITLE
How to use `update:sort` sort in v-table?

### DISCUSSION THREAD

**Discussion by iherwig** [directus/directus#17781] [Archive]
Hi, 
I have a calculated value in a v-table column by which I would like to sort the table rows. I have implemented an `update:sort` listener, which is called with the correct parameters whenever the sort icon in the column header is pressed. But sorting the v-model array in this function does not have any effect and it rather seems like sorting is done somewhere else. Is the function supposed to return a specific (e.g. boolean) value to prevent default sorting or is there anything else necessary to get this working?

**Comment by michaeltoohig** [comment]
So this is annoying and took me trial and error and asking in forums to figure out but you have to explicitly state a few things for it to work. Here's a nearly minimal example.

Note the important parts are to explicitly include the `sortable` attribute in your `headers` and to manually manage sorting by recalling the API on each `update:sort` event.


```
<template>
  <div>
    <div v-if="loading" style="margin: 25px auto; max-width: 50px">
      <v-progress-circular indeterminate />
    </div>
    <v-table
      v-else
      :headers="headers"
      :items="items"
      v-model:sort="sort"
      @update:sort="updateSort"
    />
  </div>
</template>

<script>
import { defineComponent } from 'vue';

export default defineComponent({
  inject: ['api'],
  data() {
    return {
      loading: true,
      items: undefined,
      sort: {
        by: 'date_created',
        desc: true,
      },
      headers: [
        { sortable: true, width: 180, text: 'Created', value: 'date_created' },
        { sortable: true, width: 180, text: 'Name', value: 'name' },
      ],
    }
  },
  methods: {
    buildUrl() {
      let url = '/container-tracker-endpoint';
      if (this.sort) {
        url += `?sort=${this.sort.desc ? '-' : ''}${this.sort.by}`
      }
      return url;
    },
    async fetch() {
      return new Promise((resolve, reject) => {
        const url = this.buildUrl();
        this.api.get(url)
          .then((resp) => {
            resolve(resp.data)
          })
          .catch((err) => reject(err));
      });
    },
    async updateSort(sort) {
      this.sort = sort;
      this.items = await this.fetch();
    },
  },
  async mounted() {
    this.items = await this.fetch();
    this.loading = false;
  },
}
```

**Reply by iherwig** [reply]
Thanks for this clear and instructive example.
